internal/logutil/log.go

```diff
@@ -60,3 +61,11 @@ type ctxLogKeyType struct{}
 // CtxLogKey is the key to retrieve logger from context.
 // It can be assigned to another value.
 var CtxLogKey interface{} = ctxLogKeyType{}
+
+// AssertWarn panics when in testing mode, and logs a warning msg otherwise.
+func AssertWarn(logger *zap.Logger, msg string, fields ...zap.Field) {
+	if testing.Testing() {
```

Comment:
This condition is never "true" when running with bazel.
It's ok if it's used in client-go's test only.

Reply:
Interesting, I tired to compile the following code by `go test -c ...` and run the binary, it also showed true.
```go
func TestDemo(t *testing.T) {
	fmt.Println(">>> in test?", testing.Testing())
}
```

Although the function is in internal package, I want it works in tidb ut. I'll figure out how bazel build/run test binaries.

Reply:
FYI the only work way I find is TiDB pass the `intest.InTest` to client-go via an interface, not zero cost, and greatly impact the the client's interface.
If you use intest flag condition build in client-go, bazel compile will fail...

Reply:
https://github.com/bazelbuild/rules_go/issues/3686 I realize the issue has be reported by @tiancaiamao . Maybe just let it only works in client-go ut by now.

Reply:
LGTM